pkg/ast/pipesearch/searchHandler.go

```diff
@@ -593,6 +603,149 @@ func RunQueryForNewPipeline(conn *websocket.Conn, qid uint64, root *structs.ASTN
 	}
 }
 
+func getFinalBuckets(complete *structs.PipeSearchCompleteResponse, timeRange *dtu.TimeRange) error {
```

Comment:
1) This func is 150+ lines, suggest breaking it up
2) This func is not "gettting" anything but rather it is filling up the response, so perhaps name it differently, usually I follow `actionNoun` pattern, for eg:  `fillAllBuckets` or `populateMissingBuckets`
3) nit: rename `complete` to be `completeResp`